lib/stores/S3Store.ts

```diff
@@ -374,6 +356,48 @@ export default class S3Store extends DataStore {
     this.cache.delete(id)
   }
 
+  private calcOptimalPartSize(size: number): number {
+    let optimalPartSize: number
+
+    // When upload is smaller or equal to PreferredPartSize, we upload in just one part.
+    if (size <= this.preferredPartSize) {
+      optimalPartSize = size
+    }
+    // Does the upload fit in MaxMultipartParts parts or less with PreferredPartSize.
+    else if (size <= this.preferredPartSize * this.maxMultipartParts) {
+      optimalPartSize = this.preferredPartSize
+      // Prerequisite: Be aware, that the result of an integer division (x/y) is
```

Comment:
I think we have to double check if the divisions in the lines below are actually safe. I see that you copied the code from tusd. Go has proper integer division, where the result will always be an integer. In JavaScript, that is not the case.